core/arch/aarch64/codec.c

```diff
@@ -1966,6 +1966,25 @@ encode_opnd_s10(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
     return encode_opnd_vector_reg(10, 2, opnd, enc_out);
 }
 
+/* isz: Vector element width for SIMD instructions. */
+
+static inline bool
+decode_opnd_isz(uint enc, int opcode, byte *pc, OUT opnd_t *opnd)
+{
+    uint bits = enc >> 22 & 3;
+    *opnd = opnd_create_immed_int(bits, OPSZ_2b);
+    return true;
+}
+
+static inline bool
+encode_opnd_isz(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
+{
+    if (opnd_get_immed_int(opnd) < ISZ_BYTE || opnd_get_immed_int(opnd) > ISZ_DOUBLE)
```

Comment:
> Since opnd_get_immed_int might be a function, I would avoid using it three times here and instead put uint bits = opnd_get_immed_int(opnd).

Sometimes I tend to have too much faith in compilers. I've changed it.

> Also, since line 1984 wants bits to be 0, 1, 2 or 3, I'm not sure the use of ISZ_BYTE and ISZ_DOUBLE is helpful here, so I would have put if (bits > 3), but that's just my opinion and I wouldn't be surprised if some people would disagree.

I wasn't sure, but yes, they don't add too much value I think. Ideally, I think there would be an enum, shared with the FP ops. I could do that as a follow up.